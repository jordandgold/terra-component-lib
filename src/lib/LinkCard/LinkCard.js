import React from "react";
import "./LinkCard.scss";
import PropTypes from "prop-types";
import Icon from "../Icon/Icon";

const LinkCard = ({ title, text, link }) => {
  return (
    <article className="ter-link-card">
      <h4 className="ter-link-card__title">{title}</h4>
      <p className="ter-link-card__text">{text}</p>
      <a className="ter-link-card__link" url={link.url}>
        {link.text}
        <Icon type="enclosed-arrow-right-dark-32px" size="32px" />
      </a>
    </article>
  );
};

export default LinkCard;

LinkCard.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired
};
