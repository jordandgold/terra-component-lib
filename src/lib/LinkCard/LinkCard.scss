@import "../global-styles/terra.scss";

.ter-link-card {
  &__title {
    margin-bottom: spacing("xs");
  }
  &__text {
    font-size: font-size("sm");
    margin-bottom: spacing("xs");
  }

  &__link {
    font-size: font-size("sm");
    font-weight: font-weight("bold");
  }

  .ter-icon {
    fill: color("ui-1");
    margin-left: spacing("mini");
    transform: translateY(3px);
  }
}
