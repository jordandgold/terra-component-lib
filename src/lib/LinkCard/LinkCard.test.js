import React from "react";
import LinkCard from "./LinkCard";
import { shallow } from "enzyme";

describe("LinkCard", () => {
  let wrapper;

  beforeEach(() => {
    const mockLink = { url: "#", text: "Sub-CTA" };

    wrapper = shallow(
      <LinkCard
        title="Good design is innovative"
        text="Quis nulla in officia veniam exercitation quis deserunt exercitation. Ea minim irure nostrud ipsum incididunt minim dolore elit ut eu esse ut officia."
        link={mockLink}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
