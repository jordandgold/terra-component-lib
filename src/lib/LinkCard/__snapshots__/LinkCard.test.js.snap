// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCard should match the snapshot 1`] = `
<article
  className="ter-link-card"
>
  <h4
    className="ter-link-card__title"
  >
    Good design is innovative
  </h4>
  <p
    className="ter-link-card__text"
  >
    Quis nulla in officia veniam exercitation quis deserunt exercitation. Ea minim irure nostrud ipsum incididunt minim dolore elit ut eu esse ut officia.
  </p>
  <a
    className="ter-link-card__link"
    url="#"
  >
    Sub-CTA
    <Icon
      size="32px"
      type="enclosed-arrow-right-dark-32px"
    />
  </a>
</article>
`;
